cmd/formula-analytics.rb

```diff
@@ -355,59 +355,75 @@ def influx_analytics(args)
 
     odie "No InfluxDB credentials found in HOMEBREW_INFLUXDB_TOKEN!" unless ENV["HOMEBREW_INFLUXDB_TOKEN"]
 
-    # TODO: attempt to strip out the date and ideally index columns
-    # TODO: attempt to sort on server-side
-    # TODO: handle other dimensions
-    query = <<~EOS
-      from(bucket: "analytics")
-        |> range(start: -31d, stop: now())
-        |> filter(fn: (r) => r["_measurement"] == "formula_install")
-        |> group(columns: ["package_and_options"])
-        |> sum()
-    EOS
-    result = influxdb_client.create_query_api.query_raw(query: query)
-    # remove datatype, group, empty, column name lines
-    lines = result.lines.drop(4)
-    json = {
-      category:    :formula_install,
-      total_items: 0,
-      start_date:  nil,
-      end_date:    nil,
-      total_count: 0,
-      items:       [],
-    }
-
-    lines.each do |line|
-      _, _, _index, start_date, end_date, count, name = line.split(",")
-      next if name.blank?
-
-      count = count.to_i
-
-      json[:total_items] += 1
-      json[:start_date] ||= DateTime.parse(start_date)
-      json[:end_date] ||= DateTime.parse(end_date)
-      json[:total_count] += count
-
-      json[:items] << {
-        number:  nil,
-        # TODO: handle other dimensions
-        formula: name.chomp,
-        count:   count,
-      }
+    days_ago = (args.days_ago || 30).to_i
+    if days_ago > 365
+      opoo "Analytics are only retained for 1 year, setting `--days-ago=365`."
+      days_ago = 365
     end
 
-    json[:items].sort_by! do |item|
-      -item[:count]
-    end
+    categories = []
+    categories << :formula_install if args.install?
+    categories << :cask_install if args.cask_install?
+    categories << :build_error if args.build_error?
+    categories << :os_versions if args.os_version?
 
-    json[:items].each_with_index do |item, index|
-      item[:number] = index + 1
+    field = "package"
+    tag = "pkg"
 
-      percent = (item[:count].to_f / json[:total_count]) * 100
-      item[:percent] = format_percent(percent)
-    end
+    # TODO: handle on_request and OS version dimensions
```

Comment:
@MikeMcQuaid is there anything else I'm missing here? I see some mention of `core-` analytics, but I'm not sure what that does differently. I'd also like to point out that in the InfluxDB analytics linux and mac data is merged.
```suggestion
    # TODO: handle on_request dimensions
```

Reply:
@SMillerDev basically the test here is: do https://github.com/Homebrew/homebrew-formula-analytics/blob/master/cmd/generate-analytics-api.rb and https://github.com/Homebrew/formulae.brew.sh/blob/c370cd566dc5e9d3bda616705213a506ad63c6a2/Rakefile#L48-L50 work as expected and generate/change all the data that you'd expect them too. If so: this works and this comment can be removed.

Note: we may want to think about adding an analytics token to the secrets for this repository and actually attempting to generate analytics here to avoid changes here breaking generation.

Alternatively, I think there's an argument that literally this entire repository could potentially be moved into formulae.brew.sh.

Reply:
The read-only token is in 1password for this. The only thing that isn't currently working is the argument to filter out non-brew taps but I think we can live without that for now.

Reply:
@SMillerDev yeh, that seems fine to not have.